c7n/resources/dms.py

```diff
@@ -256,157 +256,160 @@ class ModifyDmsEndpoint(BaseAction):
             'CompressionType': settings.get(
-                'compressiontype', e.get('S3Settings', {}).get(
-                    'CompressionType')) or 'none'}
-
-        if settings.get('serviceaccessrolearn'):
-            params['S3Settings']['ServiceAccessRoleArn'] = settings.get(
-                'serviceaccessrolearn')
-        if settings.get('bucketfolder'):
-            params['S3Settings']['BucketFolder'] = settings.get(
-                'bucketfolder')
-        if settings.get('externaltabledefinition'):
-            params['S3Settings']['ExternalTableDefinition'] = settings.get(
-                'externaltabledefinition')
+                'CompressionType', e['S3Settings'].get(
+                    'CompressionType')) or 'none',
+            'ExternalTableDefinition': settings.get(
+                'ExternalTableDefinition', e['S3Settings'].get(
+                    'ExternalTableDefinition', '')),
+            'ServiceAccessRoleArn': settings.get(
+                'ServiceAccessRoleArn', e['S3Settings'].get(
+                    'ServiceAccessRoleArn')),
+            'BucketFolder': settings.get(
+                'BucketFolder', e['S3Settings'].get('BucketFolder', ''))
+        }
         return params
 
     def configure_dynamodb_params(self, e, params):
-        settings = self.data.get('dynamodbsettings')
+        settings = self.data.get('DynamoDbSettings')
         if not settings:
-            raise KeyError('dynamodbsettings not provided')
+            raise KeyError('DynamoDbSettings not provided')
 
-        rolearn = self.data.get(
-            'serviceaccessrolearn',
-            e['DynamoDbSettings']['ServiceAccessRoleArn'])
-        params['DynamoDbSettings']['ServiceAccessRoleArn'] = rolearn
+        params['DynamoDbSettings'] = {
+            'ServiceAccessRoleArn': settings.get(
+                'ServiceAccessRoleArn',
+                e['DynamoDbSettings']['ServiceAccessRoleArn'])}
         return params
 
     def configure_mongodb_params(self, e, params):
-        settings = self.data.get('mongodbsettings')
+        settings = self.data.get('MongoDbSettings')
         if not settings:
-            raise KeyError('mongodbsettings not provided')
+            raise KeyError('MongoDbSettings not provided')
 
         nest = settings.get(
-            'nestinglevel', e['MongoDbSettings']['NestingLevel']).lower()
-        auth = settings.get('authtype', e['MongoDbSettings']['AuthType'])
+            'NestingLevel', e['MongoDbSettings']['NestingLevel'])
+        auth = e['MongoDbSettings']['AuthType']
         params['MongoDbSettings'] = {
             'ServerName': settings.get(
-                'servername', e['MongoDbSettings']['ServerName']),
-            'Port': settings.get('port', e['MongoDbSettings']['Port']),
+                'ServerName', e['MongoDbSettings']['ServerName']),
+            'Port': settings.get('Port', e['MongoDbSettings']['Port']),
             'NestingLevel': nest,
```

Comment:
Valid values are string 'none' or 'one'.  In this case it could possibly be `None`